### PR TITLE
Add support for Jackson2.4 to jaxrs-clients

### PR DESCRIPTION
**Pull request by uschi2000**
Thoughts, @pnepywoda

### CODE REVIEW

File: jaxrs-clients/src/integrationTests/java/com/palantir/remoting/jaxrs/Jackson24Test.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+public final class Jackson24Test {
+
+    @Rule
+    public final MockWebServer server = new MockWebServer();
+
+    // Note that the gradle setup of the integrationTests configuration forces Jackson 2.4 for this project.
+    // This construction does not work in IntelliJ since different configurations have the same dependencies
+    // (, Jackson 2.6 in this case).
+    @Test
+    public void test_CanBuildClientWithJackson24() throws JsonProcessingException {
```
Comment (uschi2000) [inline jaxrs-clients/src/integrationTests/java/com/palantir/remoting/jaxrs/Jackson24Test.java]: verified that this test fails without the fix in FeignJaxRsClientBuilder

File: jaxrs-clients/build.gradle [hunk 2]
```diff
@@ -30,4 +33,17 @@ dependencies {
     testCompile libs.hamcrest
     testCompile libs.junit
     testCompile libs.mockwebserver
+
+    processor libs.immutables
+}
+
+[configurations.integrationTestsCompile, configurations.integrationTestsCompileClasspath, configurations.integrationTestsCompileOnly, configurations.integrationTestsRuntime].each {
```
Comment (uschi2000) [inline jaxrs-clients/build.gradle]: verified (via Gradle :dependencies task) that these configurations have jackson 2.4 only

File: jaxrs-clients/src/main/java/com/palantir/remoting/jaxrs/FeignJaxRsClientBuilder.java [hunk 3]
```diff
@@ -107,12 +108,24 @@ private Contract createContract() {
...
+                ? new JacksonEncoder(objectMapper)
+                : new Jackson24Encoder(objectMapper);
+        return new InputStreamDelegateEncoder(new TextDelegateEncoder(jacksonEncoder));
+    }
+
+    // Uses reflection to determine if Jackson >= 2.5 is on the classpath by checking for the existence of the
+    // ObjectMapper#writerFor method.
+    private boolean hasJackson25() {
+        try {
+            ObjectMapper.class.getMethod("writerFor", Class.class);
```
Comment (pnepywoda) [inline jaxrs-clients/src/main/java/com/palantir/remoting/jaxrs/FeignJaxRsClientBuilder.java]: more specifically, it's going to try and use writerFor(JavaType)....but that's just a nit

Comment (uschi2000) [inline jaxrs-clients/src/main/java/com/palantir/remoting/jaxrs/FeignJaxRsClientBuilder.java]: changed.

**Comment by uschi2000** [comment]
anything else, Paul?

**Comment by pnepywoda** [comment]
+2

**Pull request merged**